src/plugins/ui_stateful/selection.ts

```diff
@@ -580,15 +580,29 @@ export class GridSelectionPlugin extends UIPlugin {
 
     const toRemove = isBasedBefore ? cmd.elements.map((el) => el + thickness) : cmd.elements;
     let currentIndex = cmd.base;
+
+    const resizing_groups: Record<number, number[]> = {};
+
     for (const element of toRemove) {
       const size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, element);
+      const current_size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
+      if (size != current_size) {
+        if (size in resizing_groups) {
+          resizing_groups[size].push(currentIndex);
+        } else {
+          resizing_groups[size] = [currentIndex];
+        }
```

Comment:
A slightly shorter way of writing the same thing
```suggestion
        resizing_groups[size] ??= [];
        resizing_groups[size].push(currentIndex);
```